examples/demo-stf/src/hooks_impl.rs

```diff
@@ -34,7 +36,7 @@ impl<C: Context> ApplyBlobHooks for Runtime<C> {
     fn begin_blob_hook(
         &self,
         sequencer: &[u8],
-        raw_blob: &[u8],
+        raw_blob: &impl Read,
```

Comment:
It seems odd to pass `&impl Read` when (IIRC) the `Read` trait doesn't have any methods which can be called on `&self`. Do we need this argument at all? If so, what do we use it for?

Reply:
Changed to pass a BufReaderWithCounter which has an accumulator to be able to read again data from the buffer.